src/bin/common/parse_command.ml

```diff
@@ -0,0 +1,1157 @@
+             no_minimal_bj $ no_sat_learning $
+             no_tableaux_cdcl_in_instantiation $
+             no_tableaux_cdcl_in_theories $ sat_plugin $ sat_solver
+            ))
+
+let parse_term_opt =
+
+  let docs = s_term in
+
+  let disable_ites =
+    let doc = "Disable handling of ite(s) on terms in the backend." in
+    Arg.(value & flag & info ["disable-ites"] ~docs ~doc) in
+
+  let inline_lets =
+    let doc =
+      "Enable substitution of variables bounds by Let. The default \
+       behavior is to only substitute variables that are bound to a \
+       constant, or that appear at most once." in
+    Arg.(value & flag & info ["inline-lets"] ~docs ~doc) in
+
+  let rewriting =
+    let doc = "Use rewriting instead of axiomatic approach." in
+    Arg.(value & flag & info ["rwt"; "rewriting"] ~docs ~doc) in
+
+  let term_like_pp =
+    let doc = "Output semantic values as terms." in
+    Arg.(value & flag & info ["term-like-pp"] ~docs ~doc) in
+
+  Term.(ret (const mk_term_opt $
+             disable_ites $ inline_lets $ rewriting $ term_like_pp
+            ))
+
+let parse_theory_opt =
+
+  let docs = s_theory in
+
+  let disable_adts =
+    let doc = "Disable Algebraic Datatypes theory." in
+    Arg.(value & flag & info ["disable-adts"] ~docs ~doc) in
+
+  let inequalities_plugin =
+    let doc =
+      "Use the given module to handle inequalities of linear arithmetic." in
+    Arg.(value & opt string (get_inequalities_plugin ()) &
+         info ["inequalities-plugin"] ~docs ~doc) in
+
+  let no_ac =
+    let doc = "Disable the AC theory of Associative and \
+               Commutative function symbols." in
+    Arg.(value & flag & info ["no-ac"] ~docs ~doc) in
+
+  let no_contracongru =
+    let doc = "Disable contracongru." in
+    Arg.(value & flag & info ["no-contracongru"] ~docs ~doc) in
+
+  let no_fm =
+    let doc = "Disable Fourier-Motzkin algorithm." in
+    Arg.(value & flag & info ["no-fm"] ~docs ~doc) in
+
+  let no_nla =
+    let doc = "Disable non-linear arithmetic reasoning (i.e. non-linear \
+               multplication, division and modulo on integers and rationals). \
+               Non-linear multiplication remains AC." in
+    Arg.(value & flag & info ["no-nla"] ~docs ~doc) in
+
+  let no_tcp =
+    let doc = "Deactivate BCP modulo theories." in
+    Arg.(value & flag & info ["no-tcp"] ~docs ~doc) in
+
+  let no_theory =
+    let doc = "Completely deactivate theory reasoning." in
+    Arg.(value & flag & info ["no-theory"] ~docs ~doc) in
+
+  let restricted =
+    let doc =
+      "Restrict set of decision procedures (equality, arithmetic and AC)." in
+    Arg.(value & flag & info ["restricted"] ~docs ~doc) in
+
+  let tighten_vars =
+    let doc = "Compute the best bounds for arithmetic variables." in
+    Arg.(value & flag & info ["tighten-vars"] ~docs ~doc) in
+
+  let use_fpa =
+    let doc = "Enable support for floating-point arithmetic." in
+    Arg.(value & flag & info ["use-fpa"] ~docs ~doc) in
```

Comment:
Does this option requires to specify the fpa theory used or does the user has to also add the fpa theory used on the command-line ?